### PR TITLE
Run tests only if go file changes

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

Making it easier to contribute to anything that is not go.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by ArthurSens** [comment]
I'm just not sure if making changes to jsonnet requires us to run tests or not 🤔 Maybe it could break e2e tests?

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ on:
       - 'main'
     tags:
       - 'v*'
+    paths:
+      - '**.go'
```
Comment (simonpasquier) [inline .github/workflows/e2e.yaml:12]: Will a change to the GitHub actions workflow trigger the e2e tests?

Comment (ArthurSens) [reply .github/workflows/e2e.yaml:12]: good catch

Comment (ArthurSens) [reply .github/workflows/e2e.yaml]: It should also be on pull_request, not on push

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 name: e2e
 on:
   pull_request:
+    paths:
+      - '**.go'
+      - '.github/workflows/e2e.yml'
```
Comment (simonpasquier) [inline .github/workflows/e2e.yaml:6]: also go.{mod,sum} files?

Comment (ArthurSens) [reply .github/workflows/e2e.yaml:6]: Only on the root level? Or all go.{mod|sum}?

Comment (simonpasquier) [reply .github/workflows/e2e.yaml:6]: probably all files for safety.

**Comment by ArthurSens** [comment]
Ah interesting, the go-related CI is required, making this change impossible 😅

**Pull request closed without merging**